### ISSUE TITLE
Should Rails 5.2.0 be tied to 0.20.x releases of Thor

### ISSUE THREAD

**Issue by jrgriffiniii** [rails/rails#32955]
### Steps to reproduce

Given that I've installed `thor` at release 0.18.1...
```bash
bundle show thor
.rvm/gems/ruby-2.5.1/gems/thor-0.18.1
```

### Expected behavior
The default Rails App. generator should succeed:
```
Rails::Generators::AppGenerator.start(['internal', '--skip_spring'])
      create
      create  README.md
      create  Rakefile
      create  .ruby-version
      create  config.ru
      create  .gitignore
      create  Gemfile
         run  git init from "."
...
```

### Actual behavior
```
Rails::Generators::AppGenerator.start(['internal', '--skip_spring'])
...
Could not find "README.md" in any of your source paths. Your current source paths are: 
.rvm/gems/ruby-2.5.1/gems/railties-5.2.0/lib/rails/generators/rails/app/templates
```

Note that this appears to have been triggered within the invocation of `#copy_file` where .tt templates are *not* automatically handled with Thor 18.x releases here:
https://github.com/rails/rails/blob/5-2-stable/railties/lib/rails/generators/rails/app/app_generator.rb#L50

It was remedied by updating to 0.19.4 and 0.20.0.

### System configuration
**Rails version**:
5.2.0
**Ruby version**:
2.5.1

**Comment by guilleiguaran** [comment]
Thor 0.19 caused this other error: https://github.com/rails/rails/issues/32726

**Comment by y-yagi** [comment]
Fixed with e10fd57e98c87774a1dc38761a7d68edee638fdf  and 0bfe5fa33d065e8b98efd6215aead6059d1a3562.
